src/main/java/org/candlepin/controller/ConsumerMigration.java

```diff
@@ -0,0 +1,115 @@
+import org.candlepin.model.IdentityCertificateCurator;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Streams;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+/**
+ * Class responsible for migration of consumers from anonymous orgs.
+ */
+public class ConsumerMigration {
+
+    private static final Logger log = LoggerFactory.getLogger(ConsumerMigration.class);
+
+    private final OwnerCurator ownerCurator;
+    private final ConsumerCurator consumerCurator;
+    private final IdentityCertificateCurator identityCertificateCurator;
+    private final ContentAccessCertificateCurator contentAccessCertificateCurator;
+    private final CertificateSerialCurator serialCurator;
+    private final int batchSize;
+
+    @Inject
+    public ConsumerMigration(OwnerCurator ownerCurator, ConsumerCurator consumerCurator,
+        IdentityCertificateCurator identityCertificateCurator,
+        ContentAccessCertificateCurator contentAccessCertificateCurator,
+        CertificateSerialCurator serialCurator, Configuration config) {
+        this.ownerCurator = Objects.requireNonNull(ownerCurator);
+        this.consumerCurator = Objects.requireNonNull(consumerCurator);
+        this.identityCertificateCurator = Objects.requireNonNull(identityCertificateCurator);
+        this.contentAccessCertificateCurator = Objects.requireNonNull(contentAccessCertificateCurator);
+        this.serialCurator = Objects.requireNonNull(serialCurator);
+        this.batchSize = config.getInt(ConfigProperties.CONSUMER_MIGRATION_BATCH_SIZE);
+    }
+
+    /**
+     * Migrates all consumers of the given anonymous org to the target org.
+     *
+     * @param originOwnerKey key of the source anonymous org
+     * @param destinationOwnerKey key of the destination org
+     */
+    public void migrate(String originOwnerKey, String destinationOwnerKey) {
+        Owner originOwner = this.ownerCurator.getByKey(originOwnerKey);
+        Owner destinationOwner = this.ownerCurator.getByKey(destinationOwnerKey);
+
+        List<String> consumerIds = this.ownerCurator.getConsumerIds(originOwner).list();
+        boolean failed = false;
+        for (List<String> consumerIdsBlock : Iterables.partition(consumerIds, this.batchSize)) {
+            try {
+                this.ownerCurator.transactional(args -> {
+                    migrateBatch(consumerIdsBlock, destinationOwner);
+                    return null;
+                }).execute();
+            }
+            catch (Exception e) {
+                log.warn("Failed to migrate a batch of consumers.", e);
+                failed = true;
+            }
+        }
+
+        if (failed) {
+            throw new ConsumerMigrationFailedException("Failed to migrate consumers");
```

Comment:
This will need to get translated into a `JobExecutionException` at the job level (or, if this method is merged into the job class, you would throw a `JobExecutionException` directly), with terminal=false, so the intent is clear for such failures

Reply:
hmmm I also see the same code for this: the job `execute()` method isn't translating this exception into a JobExecutionException

Reply:
@Januson Once this one is done, I can merge the PR

Reply:
Done.